<launch>
  <arg name="device"          default="/tmp/ttyUR"    doc="MODBUS Serial device, when using socat with the the RS-485 URCap"/>
  <arg name="gripper"         default="rg2"/>
  <arg name="changer_addr"    default="65"/>
  <arg name="dummy"           default="false"/>

  <param name="/onrobot/device"           value="$(arg device)" />
  <param name="/onrobot/gripper"          value="$(arg gripper)" />
  <param name="/onrobot/changer_addr"     value="$(arg changer_addr)" />
  <param name="/onrobot/dummy"            value="$(arg dummy)" />

  <node name="OnRobotRGStatusListener"
      ns="/onrobot_rg2"
      pkg="onrobot_rg_control"
      type="OnRobotRGStatusListener.py"
      output="log"/>

  <node name="OnRobotRGSerialNode"
        ns="/onrobot_rg2"
        pkg="ur_onrobot_rg_control"
        type="OnRobotRGSerialNode.py"
        respawn="True"
        output="log"/>

  <node name="OnRobotRGJointPositionControllerServer"
      ns="/onrobot_rg2"
      pkg="ur_onrobot_rg_control"
      type="OnRobotRGJointPositionControllerServer.py"
      output="screen"/>
</launch>
